src/lib/device/journal_vdev.cpp

```diff
@@ -637,8 +647,20 @@ bool JournalVirtualDev::Descriptor::is_offset_at_last_chunk(off_t bytes_offset)
     return false;
 }
 
+//
+// This API is ways called in single thread
+//
 void JournalVirtualDev::Descriptor::high_watermark_check() {
-    if (resource_mgr().check_journal_size(used_size(), size())) {
+#if 0
+    // high watermark check for the individual journal descriptor;
+    if (resource_mgr().check_journal_descriptor_size(used_size())) {
+        // the next resource manager audit will call truncation for this descriptor;
+        set_ready_for_truncate();
+    }
+#endif
+
+    // high watermark check for the entire journal vdev;
+    if (resource_mgr().check_journal_vdev_size(m_vdev.used_size(), m_vdev.size())) {
```

Comment:
it is via `m_journal_vdev_exceed_cb` registered from here: https://github.com/yamingk/HomeStore/blob/yk_repl_new/src/lib/device/journal_vdev.cpp#L58

Reply:
Oh ok,  it is a bit wired as the code path here seems like 
if ( DETECTION)) {
    update_metrics
    do callback to trigger action
}

however the m_event_cb is nullptr and the actual callback is done in the detection part(`(resource_mgr().check_journal_vdev_size`) .  

Can I read this as your TODO to move to m_event_cb?

Reply:
Yes. The goal is in resource mgr to move all layer_spcecif_registered_cb to general event_cb. Here in this journal vdev, I think the only consumer of log store and we can see if there is still any use case this event_cb is still needed since most of the job is done via resource manager, now.